web/annotation_layer_builder.css

```diff
@@ -77,6 +77,22 @@
   border: 1px solid transparent;
 }
 
+.annotationLayer .textWidgetAnnotation input.comb {
+  font-family: monospace;
+  padding-left: 2px;
+  padding-right: 0;
+}
+
+.annotationLayer .textWidgetAnnotation input.comb:focus {
+  /*
+   * Letter spacing is placed on the right side of each character. Hence, the
+   * letter spacing of the last character may be placed outside the visible
+   * area, causing horizontal scrolling. We avoid this by extending the width
+   * when the element has focus and revert this when it loses focus.
+   */
+  width: 115%;
```

Comment:
I agree, but after searching for a couple of hours there really isn't another way with pure CSS only. The real problem is that letter spacing is always applied on the right side of a character, whereas we would want to tune that. Since that's not possible, we'll need this workaround. There are JS-based solutions, but they are even uglier and require much more code, so this was the cleanest way to go.